app/views/repositories/empty.html.erb

```diff
@@ -33,6 +33,7 @@ See doc/COPYRIGHT.rdoc for more details.
              message: l('repositories.errors.empty_repository')) %></p>
   </div>
 </div>
+<%= render partial: 'checkout_instructions', locals: { repository: @repository, instructions: @instructions } %>
```

Comment:
I agree, we have a common pattern (e..g, in work packages) to display this larger notification / block with the empty 'warning'. We could apply this here, too, and then I'd agree to move the checkout above it.

However this is rather out of scope here.